### PR TITLE
Add nightly to version output

### PR DESCRIPTION
**Pull request by benjgil**
- add latest nightly release to `bvm version` printout (only when nightly is set as the release type)
- added version type for latest installed and currently used versions in the above printout

### CODE REVIEW

File: teambit.bvm/commands/version/version.ts [hunk 1]
```diff
@@ -128,22 +146,33 @@ function formatOutput(versions: VersionsResult): string {
...
+    )
+    {
+      return `new ${versionToCheck.releaseType?.toString() ?? ""} version (${versionToCheck.version}) of ${chalk.cyan("bit")} `+
+           `is available, upgrade your ${chalk.cyan("bit")} by running "${chalk.cyan(commandToRun)}"\n`;
+    }
   }
+  latestRemoteStableVersion = undefined;
+  const output = [newVersionAvailableText(latestRemoteStableVersion), newVersionAvailableText(latestRemoteNightlyVersion)].join("");
+    
+  return output;
```
Comment (zkochan) [inline teambit.bvm/commands/version/version.ts:177]: we should probably write some snapshot tests for these output generating functions.

File: teambit.bvm/list/gcp/gcp-list.ts [hunk 2]
```diff
@@ -18,6 +18,7 @@ export enum ReleaseTypeFilter {
   DEV = 'dev',
   NIGHTLY = 'nightly',
   STABLE = 'stable',
+  ALL = 'all'
```
Comment (zkochan) [inline teambit.bvm/list/gcp/gcp-list.ts:21]: I would suggest to not add new fields to this type. Instead create a new function like listAll() or pass an option to the function.

File: teambit.bvm/commands/version/version.ts [hunk 3]
```diff
@@ -61,19 +62,27 @@ export async function showAllVersions(options: ShowVersionsOptions = defaultShow
...
-
-  const currentVersion = config.getDefaultLinkVersion();
-  const latestInstalledVersion = (await listLocal()).latest().version;
-  const latestRemoteVersion = actualOpts.includeRemote
-    ? (await listRemote()).latest().version
+    
+  const remoteVersionsList = await listRemote({releaseType: ReleaseTypeFilter.ALL});
+  const remoteVersionsListMap = remoteVersionsList.toMap();
+  const currentVersionString = config.getDefaultLinkVersion();
+  const currentVersion = currentVersionString ? new Version( currentVersionString, remoteVersionsListMap.get(currentVersionString)) : undefined;
```
[the remark this answers is not in the document]
Comment (benjgil) [reply teambit.bvm/commands/version/version.ts:69]: done

File: teambit.bvm/commands/version/version.ts [hunk 4]
```diff
@@ -61,19 +62,27 @@ export async function showAllVersions(options: ShowVersionsOptions = defaultShow
...
+  const releaseType = config.getReleaseType();
+  const latestInstalledVersion = (await listLocal()).latest();
+  const latestRemoteStableVersion = actualOpts.includeRemote
+    ? remoteVersionsList.versionsByReleaseType([ReleaseType.STABLE]).latest()
+    : undefined;
+  const latestRemoteNightlyVersion = releaseType === ReleaseTypeFilter.NIGHTLY
```
[the remark this answers is not in the document]
Comment (benjgil) [reply teambit.bvm/commands/version/version.ts:75]: good point, done

File: teambit.bvm/commands/version/version.ts [hunk 5]
```diff
@@ -2,15 +2,16 @@ import { CommandModule, Argv } from "yargs";
...
 export type VersionsResult = {
   currentBvmVersion?: string;
   latestBvmRemoteVersion?: string;
-  currentVersion?: string;
-  latestRemoteVersion?: string;
-  latestInstalledVersion?: string;
+  currentVersion?: Version;
+  latestRemoteStableVersion?: Version;
+  latestRemoteNightlyVersion?: Version;
+  latestInstalledVersion?: Version;
```
[the remark this answers is not in the document]
Comment (benjgil) [reply teambit.bvm/commands/version/version.ts:14]: good plan, makes it easier to deal with each separately

File: teambit.bvm/commands/version/version.ts (hunk 1 above)
Comment (benjgil) [reply teambit.bvm/commands/version/version.ts:177]: They're just the existing code turned into a function to save on re-use. Isn't a snapshot test for that a bit overkill?

We should probably have a unittest for the whole functionality but bvm isn't well tested (inc this file) and it's probably not in the scope of this task to remedy that...

File: teambit.bvm/list/gcp/gcp-list.ts (hunk 2 above)
Comment (benjgil) [reply teambit.bvm/list/gcp/gcp-list.ts:21]: IMO 'ALL' is a good filter. You don't always want to filter for a specific type of release so this leaves the option to pass an enum that returns all release types

File: teambit.bvm/list/gcp/gcp-version.ts [hunk 6]
```diff
@@ -8,21 +9,22 @@ export class GcpVersion {
...
   get url() {
     return `${this.protocol}://${this.host}/${this.fileName}`;
   }
 
   /** is version is stable version  */
-  get stable() {
-    if (this.metadata?.stable) return true;
-    return false;
+  get calculatedReleaseType() {
+    const type = this.metadata?.stable ? ReleaseType.STABLE : this.releaseType ?? ReleaseType.NIGHTLY; // TODO @giladshoam what's the metadata doing here? Can we get rid of it and just use releaseType when storing the version?
```
Comment (zkochan) [inline teambit.bvm/list/gcp/gcp-version.ts:22]: do we want to use NIGHTLY by default?

**Review by zkochan — APPROVED** [review]

**Review by GiladShoham — APPROVED** [review]

**Pull request merged**